src/api/app/controllers/webui/search_controller.rb

```diff
@@ -161,9 +162,19 @@ def search
                                 issue_tracker_name: @search_tracker)
     @results = search.search(page: params[:page], per_page: @per_page)
     flash[:notice] = 'Your search did not return any results.' if @results.empty?
+  rescue ThinkingSphinx::OutOfBoundsError
+    flash[:error] = 'You are trying to access results out of bounds. Please adjust your "page" parameter accordingly.'
+    @results = []
   rescue ThinkingSphinx::SphinxError => e
     flash[:error] = "There has been an error performing the search. We're working to fix it. Please, try again later."
     Airbrake.notify(e)
     @results = []
   end
+
+  def validate_page
+    return unless params[:page].present? && params[:page] !~ /\A\d+\z/
+
+    flash[:error] = 'You must pass an integer for the "page" parameter.'
```

Comment:
I would just discard the page param if the page parameter is not a number...

Reply:
Sorry, what I meant was that you could just discard the error message and just redirect back without telling the user anything.

Reply:
I prefer to be verbose and tell the user what has gone wrong. Do you find a reason to hide those details to the user?

Reply:
LGTM